sdk/internal/recording/recording.go

```diff
@@ -624,11 +626,37 @@ func GetRecordingId(t *testing.T) string {
 }
 
 func GetRecordMode() string {
+	if !(recordMode == RecordingMode || recordMode == PlaybackMode || recordMode == LiveMode) {
+		recordMode = PlaybackMode
+	}
 	return recordMode
 }
 
+func findProxyCertLocation() (string, error) {
+	fileLocation, ok := os.LookupEnv("PROXY_CERT")
+	if ok {
+		return fileLocation, nil
+	}
+
+	// Couldn't be found, lets try to find it from the GOPATH location
+	var err error
+	goPath, ok := os.LookupEnv("GOPATH")
+	if !ok {
+		goPath, err = os.UserHomeDir()
+		if err != nil {
+			return "", err
+		}
+	}
+
+	goPath = filepath.Join(goPath, "src", "github.com", "Azure", "azure-sdk-for-go", "eng", "common", "testproxy", "dotnet-devcert.crt")
+	return goPath, nil
```

Comment:
If I'm reading this right, when `goPath` doesn't exist, we return the error from `ioutil.ReadFile` directly. I wonder we should add context to that message so the reader doesn't wonder what that file read was for.

Reply:
We only return the error if the `os.UserHomeDir()` cannot be found, which is if the `$HOME` envvar is not set